src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -53,8 +75,20 @@ class TestTargetSetup:
   input_files_digest: Digest
 
 
+def get_packages_to_cover(
+  test_target: PythonTestsAdaptor,
+  source_root_stripped_file_paths: Tuple[str, ...],
+) -> Set[str]:
+  if hasattr(test_target, 'coverage'):
+    return set(test_target.coverage)
+  return set(
+    os.path.dirname(source_root_stripped_source_file_path).replace(os.sep, '.')
+    for source_root_stripped_source_file_path in source_root_stripped_file_paths
+  )
+
+
 @rule
-async def setup_pytest_for_target(test_target: PythonTestsAdaptor, pytest: PyTest) -> TestTargetSetup:
+async def setup_pytest_for_target(test_target: PythonTestsAdaptor, pytest: PyTest, test_options: TestOptions) -> TestTargetSetup:
```

Comment:
Hm. We might want to avoid it because it reduces the composibility of rules...?: if you used this rule to compute pytest for some purpose other than the `test` goal (...farfetched, but possible), it would be getting its options in the wrong place.

I expect though that this is a workaround for the fact that we don't have #7490, which would enable https://github.com/pantsbuild/pants/pull/8910#discussion_r363480513 ... ie: doing the equivalent of `Get[TestResult](Params(Target, CoverageConfiguration))` ... or something. If that's the case, a comment saying: "TODO: Rather than consuming the TestOptions subsystem, the TestRunner should pass on coverage configuration via #7490".